### ISSUE TITLE
All Transactions view - Selected Transactions Amount

### ISSUE THREAD

**Issue by rmelillo76** [moneymanagerex/moneymanagerex#6233]
Currently, the total amount of the selected transactions is only shown when the user is on an individual account view. I'm not sure if there was an underlying reason we chose not to show it on All Transactions, but to me it would make sense to have it there as well.

The word 'balance' may be part of the reason/confusion. By design, there is no Balance column on the All Transactions view since it would not make sense to show a running balance when multiple accounts are displayed. Perhaps it should be labeled "Selected Transactions Total:" or "Total Amount Selected"?

Individual Account View:
![image](https://github.com/moneymanagerex/moneymanagerex/assets/95182117/1374b37e-1545-4ca1-9975-9878bef5cf95)

All Transactions View:
![image](https://github.com/moneymanagerex/moneymanagerex/assets/95182117/e4747d02-197c-484e-9f2e-de0c28bb2eff)

**Comment by whalley** [comment]
I think the primary reason was that in the 'All Transactions View' the accounts may have varying currencies. Admittedly we could use the base currency to display this.

**Comment by whalley** [comment]
@rmelillo76 Given that in the All Transactions View accounts may have differing currencies are you Ok to close this?

**Comment by rmelillo76** [comment]
> @rmelillo76 Given that in the All Transactions View accounts may have differing currencies are you Ok to close this?

Agreed - Make sense to close. Thanks for looking into this.

**Comment by splisp** [comment]
I asked it again (#7117) without realizing that it had already been done before.  I second the request: I understand the problem in implementing it, but using multiple currencies is an advanced feature to my eyes and as an average user (I think, at least!) I have all my accounts in a single currency and being able to view totals across one bank account, one credit card account and one cash account is something that seems normal.

**Comment by splisp** [comment]
> The word 'balance' may be part of the reason/confusion.

In fact, "Balance" is a bit confusing.  It's more of an algebraic sum of the selection: even when applied to a single account, it has no relationship with the balance of that account at any given date

**Comment by georgeef** [comment]
I can look at it, but I work on date ranges at the moment. Could you reopen this issue?

Regarding "Balance": I agree, it is a confusing. Maybe "Total" or "Sum" is a better term.

@whalley as FYI

**Comment by rmelillo76** [comment]
> I asked it again (#7117) without realizing that it had already been done before. I second the request: I understand the problem in implementing it, but using multiple currencies is an advanced feature to my eyes and as an average user (I think, at least!) I have all my accounts in a single currency and being able to view totals across one bank account, one credit card account and one cash account is something that seems normal.

I agree, I would think most users have their accounts in a single currency, and this would be a nice feature. Thanks!

**Comment by rmelillo76** [comment]
> I can look at it, but I work on date ranges at the moment. Could you reopen this issue?
> 
> Regarding "Balance": I agree, it is a confusing. Maybe "Total" or "Sum" is a better term.
> 
> @whalley as FYI

Re-opened, thanks @georgeef !

**Comment by whalley** [comment]
Just looking at implementing this and one further complication, other than currency, is that the money flow cannot be sensibly calculated.

For WITHDRAWALS and DEPOSITS we can sum as -ve and +ve figures respectfully.
For TRANSFERS when single accounts are selected the sum is calculated as a -ve and +ve depending on the source and destination account of the transfer.

For the "All Transactions" view we could:

(1) Ignore TRANSFERS in the summation
(2) Display a figure including transfers and one without, similar to to how it is done on the Transaction Report.

Views? I think (1) better aligns with the how the data is presented in the account view as both WITHDRAWAL and DEPOSIT values are shown - which cancel each other out when summing.

**Comment by georgeef** [comment]
A TRANSFER can be thought as two transactions: a WITHDRAWAL from the source account and a DEPOSIT to the destination account. I think it makes sense to include it in the sum as if it was two transactions. Different currencies or different amount in the two accounts is not an issue, because they can be different also in two separate transactions.

**Comment by splisp** [comment]
> A TRANSFER can be thought as two transactions: a WITHDRAWAL from the source account and a DEPOSIT to the destination account. I think it makes sense to include it in the sum as if it was two transactions.

I'd add that if both source and destination accounts are included in the report, then they cancel out; otherwise, it makes sense to only count a WITHDRAWAL or DEPOSIT transaction (provided there are possible scenarios where one of the two accounts is not included)

**Comment by whalley** [comment]
> otherwise, it makes sense to only count a WITHDRAWAL or DEPOSIT transaction (provided there are possible scenarios where one of the two accounts is not included)

I did consider this but can could find no instance where one of the accounts would be excluded. Due to the presentation of the "All Transactions" panel if one of the accounts appears in the report then they both do as the source and destination of the transfer are presented on a single line. This is different from the "Transaction Report" where multiple lines are used to represent the outflow and inflow of money , alongside category splits.

The implementation can be tested in the latest v1.9.2 Beta build here: https://ci.appveyor.com/project/guanlisheng/moneymanagerex/builds/52795488/job/dt2hj782oyfsjqdt/artifacts
